File: ProblemSet1 - C /mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt user for the pyramid's heigh
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    // Print a pyramid of that height
    for (int rowL = 0; rowL < height; rowL++)
    {
        // Check if we need to print a space or a hash (#)
        for (int columnL = 0; columnL < height; columnL++)
        {
            if (rowL + columnL < height - 1)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        // Print two spaces between the left and right pyramids to make a gap
        printf("  ");
        // Loop to print hashes in the right pyramid based on current row number
        for (int rowR = 0; rowR < rowL + 1; rowR++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: ProblemSet6 - Python/sentimental-mario-more/mario.py
from cs50 import get_int

def main():
    height = get_height("Height: ")
    pyramid(height)

# Prompt user for a valid input
def get_height(prompt):
    while True:
        h = get_int(prompt)
        # Valid input must be between 1 and 8(inclusive)
        if h > 0 and h < 9:
            return h

# Print pyramid
def pyramid(rows):
    for row in range(1, rows + 1):
        # Print left pyramid
        print(" " * (rows - row) + "#" * row, end="")
        # Print gap of two spaces between pyramids
        print("  ", end="")
        # Print right pyramid, only hashes
        print("#" * row)

if __name__ == "__main__":
    main()
